ext/legacycustomsearches/legacycustomsearches.php

```diff
@@ -95,14 +95,22 @@ function legacycustomsearches_civicrm_entityTypes(&$entityTypes) {
  *
  * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_navigationMenu
  */
-//function legacycustomsearches_civicrm_navigationMenu(&$menu) {
-//  _legacycustomsearches_civix_insert_navigation_menu($menu, 'Mailings', array(
-//    'label' => E::ts('New subliminal message'),
-//    'name' => 'mailing_subliminal_message',
-//    'url' => 'civicrm/mailing/subliminal',
-//    'permission' => 'access CiviMail',
-//    'operator' => 'OR',
-//    'separator' => 0,
-//  ));
-//  _legacycustomsearches_civix_navigationMenu($menu);
-//}
+function legacycustomsearches_civicrm_navigationMenu(&$menu) {
+  _legacycustomsearches_civix_insert_navigation_menu($menu, 'Search', [
+    'label' => E::ts('Custom Searches'),
+    'name' => 'Custom Searches',
+    'url' => 'civicrm/contact/search/custom/list?reset=1',
+    'permission' => NULL,
+    'operator' => 'OR',
+    'separator' => 0,
+  ]);
+  _legacycustomsearches_civix_insert_navigation_menu($menu, 'Administer/Customize Data and Screens', [
+    'label' => E::ts('Manage Custom Searches'),
+    'name' => 'Manage Custom Searches',
+    'url' => 'civicrm/admin/options/custom_search?reset=1',
+    'permission' => 'administer CiviCRM',
+    'operator' => 'OR',
+    'separator' => 0,
+  ]);
+  _legacycustomsearches_civix_navigationMenu($menu);
```

Comment:
FYI. We have several ways for extensions to add to the nav menu, and none of them are very good:

1. Insert during installation (`hook_civicrm_install`) using the api
2. Use `hook_civicrm_navigationMenul`
3. 2. Use `hook_civicrm_managed`

They all have downsides. 1 requires manual cleanup during uninstall, and manual handling of multi-domains. 2 is non-configurable, frustrating admins who want to customize their menus. 3 works well as long as you take care of multi-domains yourself. So on balance I think 3 is my favorite, and I've recently started using this pattern:

https://github.com/civicrm/civicrm-core/blob/5bee23815cd56c5214634ee7f40a1b8f52f0ac20/ext/search_kit/managed/Navigation_search_kit.mgd.php